### PR TITLE
Add Azure ML job CODEOWNERS

### PR DESCRIPTION
**Pull request by DouglasXiaoMS**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by TonyJ1 — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -119,6 +119,10 @@
 # PRLabel: %Local Endpoints
 /sdk/ml/azure-ai-ml/azure/ai/ml/_local_endpoints/                    @NonStatic2014 @arunsu @stanley-msft @JustinFirsching
 
+# PRLabel: %Job
```
Comment (jsquire) [inline .github/CODEOWNERS:122]: This label is way too general.   Can we please find a more specific label that better encapsulates this service?

Comment (DouglasXiaoMS) [reply .github/CODEOWNERS:122]: thanks for the comments, will modify in the future if too many pr requests
so far my team would own all pull requests related to jobs

Comment (jsquire) [reply .github/CODEOWNERS:122]: This makes repository automation and reporting difficult, as we use labels as identifiers for a lot of logic.   Let's please adjust the label name to something more specific to this service.

Comment (needuv) [reply .github/CODEOWNERS:122]: How about something like `ML Job`?

Comment (jsquire) [reply .github/CODEOWNERS:122]: perfect.   We should probably apply the same pattern to "Local Endpoints" as well to keep the pattern predictable

**Pull request merged**